livepatch-calc.c

```diff
@@ -55,20 +55,12 @@ static struct klp_patch patch = {
 
 static int livepatch_calc_init(void)
 {
-    int ret = klp_register_patch(&patch);
-    if (ret)
-        return ret;
-    ret = klp_enable_patch(&patch);
-    if (ret) {
-        WARN_ON(klp_unregister_patch(&patch));
-        return ret;
-    }
-    return 0;
+    return klp_enable_patch(&patch);
```

Comment:
Livepatch might fail, and you should take the failure into consideration.

Reply:
klp_enable_patch after v5.1 merges klp_register_patch and klp_unregister_patch.
v5.0 ( https://elixir.bootlin.com/linux/v5.0.21/source/kernel/livepatch/core.c#L884)
v5.1 ( https://elixir.bootlin.com/linux/v5.1/source/kernel/livepatch/core.c#L989 )